### PR TITLE
Kotlin: Resugar `for` loops

### PR DESCRIPTION
**Pull request by tamasvajk**

### CODE REVIEW

File: java/ql/test/kotlin/library-tests/stmts/PrintAst.expected [hunk 1]
```diff
@@ -116,59 +116,41 @@ stmts.kt:
...
+#   38|                     1: [IntegerLiteral] 100
+#   36|                   1: [BlockStmt] { ... }
+#   37|                     0: [ExprStmt] <Expr>;
+#   37|                       0: [WhenExpr] when ...
+#   37|                         0: [WhenBranch] ... -> ...
+#   37|                           0: [GTExpr] ... > ...
+#   37|                             0: [VarAccess] x
+#   37|                             1: [VarAccess] y
+#   37|                           1: [BreakStmt] break
 #   41|         4: [BlockStmt] { ... }
```
Comment (tamasvajk) [inline java/ql/test/kotlin/library-tests/stmts/PrintAst.expected:154]: This should also be converted to a `for` loop, but its structure doesn't match what we expect currently: `body.statements.size == 4`.

File: java/ql/test/kotlin/library-tests/stmts/CONSISTENCY/javaEquivalent.expected [hunk 2]
```diff
@@ -0,0 +1,3 @@
+| file://:0:0:0:0 | Couldn't find a Java equivalent function to kotlin.Int.rangeTo in java.lang.Integer |
```
Comment (tamasvajk) [inline java/ql/test/kotlin/library-tests/stmts/CONSISTENCY/javaEquivalent.expected:1]: This file should be removed. We've removed this consistency query.

File: java/kotlin-extractor/src/main/kotlin/KotlinFileExtractor.kt [hunk 3]
```diff
@@ -4016,22 +4125,34 @@ open class KotlinFileExtractor(
...
+    private fun extractLoopWithCondition(
+        loop: IrLoop,
+        stmtExprParent: StmtExprParent,
+        callable: Label<out DbCallable>
+    ) {
+        val id = extractLoop(loop, 1, stmtExprParent, callable) { parent, idx ->
+            if (loop is IrWhileLoop) {
+                val id = tw.getFreshIdLabel<DbWhilestmt>()
+                tw.writeStmts_whilestmt(id, parent, idx, callable)
+                id
```
Comment (smowton) [inline java/kotlin-extractor/src/main/kotlin/KotlinFileExtractor.kt:4148]:
```suggestion
                tw.getFreshIdLabel<DbWhilestmt>().also { 
                  tw.writeStmts_whilestmt(id, parent, idx, callable)
                }
```

File: java/kotlin-extractor/src/main/kotlin/KotlinFileExtractor.kt [hunk 4]
```diff
@@ -3174,6 +3174,92 @@ open class KotlinFileExtractor(
...
+
+        val iteratorVariable = e.statements[0] as? IrVariable
+        val innerWhile = e.statements[1] as? IrWhileLoop
+
+        if (iteratorVariable == null ||
+            iteratorVariable.origin != IrDeclarationOrigin.FOR_LOOP_ITERATOR ||
+            innerWhile == null ||
+            innerWhile.origin != IrStatementOrigin.FOR_LOOP_INNER_WHILE) {
+            return false
+        }
```
Comment (igfoo) [inline java/kotlin-extractor/src/main/kotlin/KotlinFileExtractor.kt:3208]: There is redundancy between this and the origin checks above, isn't there? i.e. We should be able to refactor as something like
```
if (e.origin != IrStatementOrigin.FOR_LOOP || e.statements.size != 2) {
    return false
}
val iteratorVariable = e.statements[0] as? IrVariable
val innerWhile = e.statements[1] as? IrWhileLoop
if (iteratorVariable?.origin != IrDeclarationOrigin.FOR_LOOP_ITERATOR) {
    return false
}
if (innerWhile?.origin != IrStatementOrigin.FOR_LOOP_INNER_WHILE) {
    return false
}
```

**Review by igfoo — APPROVED** [review]

**Pull request merged**